src/Modal.svelte

```diff
@@ -1,19 +1,58 @@
 <script>
-    export let showModal;
+    import { createEventDispatcher } from 'svelte';
+
+    const dispatch = createEventDispatcher();
+
+    export let maxNumberOfMessages;
+    export let env;
+    let currentNumberOfMessages, currentEnv;
 
     let dialog;
 
-    $: if (dialog && showModal) dialog.showModal();
+    export function showModal() {
+        if (!dialog.open) {
+            dialog.showModal();
+        }
+        currentNumberOfMessages = maxNumberOfMessages;
+        currentEnv = env;
+    }
+
+    function closeModal() {
+        if (dialog.open) {
+            dialog.close();
+        }
+    }
+
+    function applyConfig() {
+        // Let parent know about config update
+        dispatch('configChange');
+        maxNumberOfMessages = currentNumberOfMessages;
+        env = currentEnv;
+        console.log(env);
+    }
 </script>
 
-<dialog
-    bind:this={dialog}
-    on:close={() => (showModal = false)}
-    on:click|self={() => dialog.close()}
->
+<dialog bind:this={dialog} on:click|self={closeModal}>
     <div on:click|stopPropagation>
-        <slot />
-        <button on:click={() => dialog.close()}>Close</button>
+        <p>Max number of messages</p>
+        <input
+            id="messages-input-box"
+            type="number"
+            min="0"
+            bind:value={currentNumberOfMessages}
+            class:error-state={currentNumberOfMessages <= 0}
+        />
+        <p>Mode</p>
+        <label>
+            <input type="radio" bind:group={currentEnv} name="currentEnv" id="dev" value="dev" />
+            Development
+        </label>
+        <label>
+            <input type="radio" bind:group={currentEnv} name="currentEnv" id="prod" value="prod" />
+            Production
+        </label>
+        <button disabled={currentNumberOfMessages < 0} on:click={applyConfig}>Apply</button>
```

Comment:
what if we don't have an "Apply" button but apply changes as they happen in the modal?

Reply:
I can remove the Apply button if you'd like. I think it will be nice. How should I handle intermediate inputs such as 0 or an empty string for max message count?

Reply:
Hm, ignore empty values and have a timeout before you apply changes?

Reply:
Done